docs/api/v2.rst

```diff
@@ -0,0 +1,338 @@
+API v2
+======
+
+The Read the Docs API uses :abbr:`REST (Representational State Transfer)`.
+JSON is returned by all API responses including errors
+and HTTP response status codes are to designate success and failure.
+
+.. note::
+
+    A newer API v3 is in early development stages.
+    Some improvements coming in v3 are:
+
+    * Write access
+    * Simpler URLs which use slugs instead of numeric IDs
+    * Improved error reporting
+
+    If there are features you would like in v3, please get in touch
+    in the `issue tracker <https://github.com/rtfd/readthedocs.org/issues>`_.
+
+
+Authentication and authorization
+--------------------------------
+
+Requests to the Read the Docs public API are for public information only
+and do not require any authentication.
+
+
+Resources
+---------
+
+Projects
+~~~~~~~~
+
+Projects are the main building block of Read the Docs.
+Projects are built when there are changes to the code
+and the resulting documentation is hosted and served by Read the Docs.
+
+As an example, this documentation is part of the `Docs project`_
+which has documentation at https://docs.readthedocs.io.
+
+You can always view your Read the Docs projects in your `project dashboard`_.
+
+.. _Docs project: https://readthedocs.org/projects/docs/
+.. _project dashboard: https://readthedocs.org/dashboard/
+
+Project search
+++++++++++++++
+
+.. tip::
+
+    This API is in need of simplifying for end users.
+    Largely this endpoint should be used only to get the Project ID
+    which is used in other API calls.
+
+.. http:get::  /api/v2/search/project/?q={query}
```

Comment:
I'm not sure we should document this. It's not meant to be a public API, and it will be changing in our search upgrade GSOC project (#4265)

Reply:
If there's no way to get the project ID in APIv2 then the API can't really be used for projects. Is there a better way?

Reply:
I think what I'll do is just remove this call. Perhaps a nice addition (in another PR) would be to add filtering that would allow searching based on a few fields (notably `?slug=`) so people can get the ID of their project.